pandas/tools/util.py

```diff
@@ -111,8 +128,15 @@ def to_numeric(arg, errors='raise'):
         coerce_numeric = False if errors in ('ignore', 'raise') else True
 
         try:
-            values = lib.maybe_convert_numeric(values, set(),
-                                               coerce_numeric=coerce_numeric)
+            _values = lib.maybe_convert_numeric(values, set(),
+                                                coerce_numeric=coerce_numeric)
+
+            if compact_ints and issubclass(_values.dtype.type, np.integer):
+                values = lib.downcast_int64(_values, {np.int64: -1},
```

Comment:
Fair point.  I was using that as a placeholder because the `cdef` sets `int_64 NA = na_values[np.int64]`, but I could easily just leave that uninitialized.  Fixed.